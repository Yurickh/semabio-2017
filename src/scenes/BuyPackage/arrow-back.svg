<?xml version="1.0" encoding="UTF-8"?>
<svg width="17px" height="14px" viewBox="0 0 17 14" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 45.2 (43514) - http://www.bohemiancoding.com/sketch -->
    <title>Shape</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Artboard-2" transform="translate(-266.000000, -42.000000)" fill-rule="nonzero" fill="#000000">
            <g id="noun_1203396_cc" transform="translate(266.000000, 42.000000)">
                <polygon id="Shape" points="1.29200004 7.1400002 16.7960005 7.1400002 16.7960005 6.46000018 1.29200004 6.46000018 7.75200022 0 6.80000019 0 0 6.80000019 6.80000019 13.6000004 7.75200022 13.6000004"></polygon>
            </g>
        </g>
    </g>
</svg>