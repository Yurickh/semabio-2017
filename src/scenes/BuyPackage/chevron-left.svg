<?xml version="1.0" encoding="UTF-8"?>
<svg width="13px" height="25px" viewBox="0 0 13 25" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 45.2 (43514) - http://www.bohemiancoding.com/sketch -->
    <title>Shape</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Artboard-2" transform="translate(-1139.000000, -134.000000)" fill="#555555">
            <g id="Group-10" transform="translate(1002.000000, 134.000000)">
                <g id="Group-9" transform="translate(88.000000, 0.000000)">
                    <g id="noun_988402_cc" transform="translate(55.500000, 12.500000) rotate(-180.000000) translate(-55.500000, -12.500000) translate(49.000000, 0.000000)">
                        <g id="Group">
                            <path d="M12.3961795,0.0944741379 C12.3445707,0.106805003 12.2974947,0.13490073 12.2607629,0.175293103 L0.344095917,12.2442529 C0.291354484,12.298395 0.261576366,12.3729263 0.261576366,12.4507902 C0.261576366,12.5286541 0.291354484,12.6031855 0.344095917,12.6573276 L12.2607629,24.7262931 C12.3276858,24.8102094 12.4338607,24.8462729 12.5342734,24.8191942 C12.6346862,24.7921155 12.7117456,24.7066384 12.7327815,24.5990019 C12.7538174,24.4913655 12.7151446,24.3804263 12.6331587,24.3132184 L0.928080292,12.4508046 L12.6331587,0.588367816 C12.7112152,0.508386129 12.7367475,0.38669639 12.6980403,0.279135482 C12.659333,0.171574573 12.5638583,0.0989062647 12.4554243,0.0944741379 C12.4357355,0.0921757324 12.4158684,0.0921757324 12.3961795,0.0944741379 L12.3961795,0.0944741379 Z" id="Shape"></path>
                        </g>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>