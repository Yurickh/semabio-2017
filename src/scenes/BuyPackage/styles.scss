@import 'ambience/color';
@import 'ambience/responsive';
@import 'base/mask-image';
@import 'base/responsive-prop';

.page-package {
  width: 66vw;
  margin: 10vh auto;

  color: $color-emperor;

  & > .arrow {
    width: 17px;
    height: 14px;

    cursor: pointer;

    background-color: $color-emperor;

    @include mask-image('./arrow-back.svg');
  }

  & > .title {
    margin-top: 1em;

    @include responsive-prop(font-size, 30, 30);
  }

  & > .title > span {
    text-transform: uppercase;
  }

  & > .subtitle {
    display: flex;
    justify-content: space-between;

    margin-top: 1em;

    @include responsive-prop(font-size, 18, 18);
  }

  & > .subtitle > .navigator {
    display: flex;
    align-items: center;
    justify-content: space-between;

    @include responsive-prop(width, 150, 150);
  }

  & > .subtitle > .navigator > .chevron {
    width: 12px;
    height: 24px;

    cursor: pointer;

    background-color: $color-emperor;

    @include mask-image('./chevron-left.svg');
  }

  & > .subtitle > .navigator > .chevron.-back {
    transform: rotate(180deg);
  }

  & > .subtitle > .navigator > .chevron.-disabled {
    background-color: rgba($color-emperor, .4);
  }

  & > .variant-selector {
    user-select: none;

    @include responsive-prop(margin-top, 26, 26);
  }

  & > footer {
    display: flex;
    align-items: center;
    justify-content: space-between;

    @include small-only {
      flex-direction: column;

      height: 10em;
    }
  }

  & > footer > .simple-button {
    @include responsive-prop(width, 182, 182);
    @include responsive-prop(height, 55, 55);
    @include responsive-prop(font-size, 18, 18);
  }

  & > footer > .simple-button.-disabled {
    cursor: not-allowed;
    pointer-events: none;

    opacity: .4;
    background-color: $color-emperor;

    &:hover {
      background-color: transparent;
    }
  }
}
