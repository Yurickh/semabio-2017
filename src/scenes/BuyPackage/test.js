import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import BuyPackage from '../BuyPackage'

describe('BuyPackage', () => {
  it('should render correctly', () => {
    const tree = shallow(<BuyPackage
      match={{
        params: {
          product: 'E'
        }
      }}
    />)
    expect(toJSON(tree)).toMatchSnapshot()
  })
})
