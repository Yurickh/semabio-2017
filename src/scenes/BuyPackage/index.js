import React, { Component } from 'react'

import ShopClient from '../../helpers/ShopClient'
import GraphQL from '../../helpers/GraphQL'
import PackageIds from '../../helpers/PackageIds'

import Button from '../../components/Button'
import VariantSelector from '../../components/VariantSelector'

import './styles.css'

class BuyPackage extends Component {
	state = {
		selectedCourses: [],
		currentBlock: 'A',
	}

	redirectToHome = () => {
		window.location.href = '/'
	}

	componentWillMount() {
		const { product } = this.props.match.params
		const validProducts = ['B', 'C', 'D', 'E']

		if (!validProducts.includes(product)) {
			return this.redirectToHome()
		}

		this.blocks = ['A'].concat(validProducts)

		this.totalCourses = {
			'B': 1,
			'C': 1,
			'D': 2,
			'E': 3,
		}[product]

		this.currentColor = {
			color: {
				'B': '#FCB755',
				'C': '#58C166',
				'D': '#2CABCA',
				'E': '#911AEC',
			}[product]
		}
	}

	remainingCourses = () => {
		return this.totalCourses - this.state.selectedCourses.length
	}

	backVariant = () => {
		return this.state.currentBlock === this.blocks[0] ? '-disabled' : ''
	}

	nextVariant = () => {
		const [ last ] = this.blocks.slice(-1)
		return this.state.currentBlock === last ? '-disabled' : ''
	}

	refreshCourseList = (courseList) => {
		this.setState({
			selectedCourses: courseList
		})
	}

	previousBlock = () => {
		const currentIndex = this.blocks.indexOf(this.state.currentBlock)

		if (currentIndex === 0) return

		this.setState({
			currentBlock: this.blocks[currentIndex - 1]
		})
	}

	nextBlock = () => {
		const currentIndex = this.blocks.indexOf(this.state.currentBlock)

		if (currentIndex === this.blocks.length - 1) return

		this.setState({
			currentBlock: this.blocks[currentIndex + 1]
		})
	}

	nextStep = () => {
		const { product } = this.props.match.params
		const shopClient = ShopClient()

    shopClient.createCheckout()
		.then(checkout => {
			if (product !== 'B') return checkout

			const checkoutId = GraphQL.read(checkout).get('id')
			const lineItems = {
				variantId: PackageIds(product),
				quantity: 1,
			}

			return shopClient.addLineItems(checkoutId, lineItems)
		})
    .then(checkout => {
      const checkoutId = GraphQL.read(checkout).get('id')
      const lineItems = this.state.selectedCourses.map(course => ({
				variantId: course.id,
				quantity: 1,
			}))

      return shopClient.addLineItems(checkoutId, lineItems)
    })
    .then(checkout => {
			if (product === 'B') {
				return window.location.href = GraphQL.read(checkout).get('webUrl')
			}

			const checkoutId = GraphQL.read(checkout).get('id')
			window.location.href = `/comprar/${product}/${checkoutId}`
    })
	}

	render() {
		const { product } = this.props.match.params
		const { currentBlock } = this.state
		return (
			<div className="page-package">
				<div className="arrow" onClick={this.redirectToHome} />

				<h1 className="title">
					Legal! Você escolheu o <span style={this.currentColor}>pacote {product}</span>
				</h1>

				<section className="subtitle">
					{
						this.totalCourses > 0 && (
							<p>
								Escolha <b>{this.remainingCourses()}/{this.totalCourses} Minicursos</b>
							</p>)
					}
					<nav className="navigator">
						Bloco {currentBlock}
						<div className={`chevron -back ${this.backVariant()}`} onClick={this.previousBlock} />
						<div className={`chevron -next ${this.nextVariant()}`} onClick={this.nextBlock} />
					</nav>
				</section>

				<VariantSelector
					product={currentBlock}
					currentColor={this.currentColor.color}
					onChange={this.refreshCourseList}
				/>

				<footer>
					<span>ATENÇÃO: Não haverá reembolso em caso de desistência</span>
					<Button
						color='green'
						onClick={this.nextStep}
						className={!!this.remainingCourses() ? '-disabled' : ''}>
						Continuar
					</Button>
				</footer>

			</div>
		)
	}
}

export default BuyPackage
