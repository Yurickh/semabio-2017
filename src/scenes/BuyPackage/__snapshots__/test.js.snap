// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyPackage should render correctly 1`] = `
<div
  className="page-package"
>
  <div
    className="arrow"
    onClick={[Function]}
  />
  <h1
    className="title"
  >
    Legal! Você escolheu o 
    <span
      style={
        Object {
          "color": "#911AEC",
        }
      }
    >
      pacote 
      E
    </span>
  </h1>
  <section
    className="subtitle"
  >
    <p>
      Escolha 
      <b>
        3
        /
        3
         Minicursos
      </b>
    </p>
    <nav
      className="navigator"
    >
      Bloco 
      A
      <div
        className="chevron -back -disabled"
        onClick={[Function]}
      />
      <div
        className="chevron -next "
        onClick={[Function]}
      />
    </nav>
  </section>
  <VariantSelector
    allowedToSelect={true}
    currentColor="#911AEC"
    onChange={[Function]}
    product="A"
  />
  <footer>
    <span>
      ATENÇÃO: Não haverá reembolso em caso de desistência
    </span>
    <Button
      className="-disabled"
      color="green"
      onClick={[Function]}
    >
      Continuar
    </Button>
  </footer>
</div>
`;
